### PR TITLE
Fix CMake support on Win32

### PR DESCRIPTION
**Pull request by myd7349**
Hi! I am trying to add `raylib` to `vcpkg`: https://github.com/Microsoft/vcpkg/pull/5946.
And there are some problems with the CMake support on Win32:

- `find_package(PkgConfig)` causes `find_package(raylib CONFIG REQUIRED)` failed on Win32
- When built as static library, the library name is `raylib_static.lib` and `find_library(raylib_LIBRARY NAMES raylib HINTS ${${XPREFIX}_LIBRARY_DIRS})` will not find it
- When built as a shared library, `BUILD_LIBTYPE_SHARED` is necessary. Otherwise, no export library `raylib.lib` will be created since no symbols are exported
- When built as a shared library, the dll file is generally installed to the `bin` dir: https://cmake.org/cmake/help/v3.0/command/install.html?highlight=install

### CODE REVIEW

**Comment by myd7349** [comment]
The CI failed because of `LoadMesh`.
The `LoadMesh` is deleted in: https://github.com/raysan5/raylib/commit/d89d24c5e8930c18a93a6403e26914a9e2b23b84

**Comment by raysan5** [comment]
Thanks! That sounds great to me! I'm not a CMake expert but I'm sure @a3f could take a look. :)

File: cmake/raylib-config.cmake [hunk 1]
```diff
@@ -26,7 +26,7 @@ find_path(raylib_INCLUDE_DIR
 )
 
 find_library(raylib_LIBRARY
-    NAMES raylib
+    NAMES raylib raylib_static
```
Comment (a3f) [inline cmake/raylib-config.cmake]: That way you lose the ability to influence whether you get the static or shared raylib on other platforms, no?
Can you make adding `raylib_static` to `NAMES` dependent on `raylib_USE_STATIC_LIBS`?

Comment (myd7349) [reply cmake/raylib-config.cmake]: @a3f Thanks for the review! I have update the PR.

**Comment by a3f** [comment]
I've rebased your commits on top of master and It looks like your PR [breaks mingw 32 & 64 bit builds](https://ci.appveyor.com/project/a3f/raylib/builds/23701287):
```
C:\mingw-w64\i686-6.3.0-posix-dwarf-rt_v5-rev1\mingw32\bin\gcc.exe -fno-strict-aliasing -Werror=implicit-function-declaration -Werror=pointer-arith -m32 -g  -m32 -Wl,--whole-archive CMakeFiles\models_rlgl_solar_system.dir/objects.a -Wl,--no-whole-archive  -o models_rlgl_solar_system.exe -Wl,--out-implib,libmodels_rlgl_solar_system.dll.a -Wl,--major-image-version,0,--minor-image-version,0 @CMakeFiles\models_rlgl_solar_system.dir\linklibs.rsp
CMakeFiles\models_rlgl_solar_system.dir/objects.a(models_rlgl_solar_system.c.obj): In function `main':
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:81: undefined reference to `rlPushMatrix'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:82: undefined reference to `rlScalef'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:84: undefined reference to `rlPopMatrix'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:86: undefined reference to `rlPushMatrix'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:87: undefined reference to `rlRotatef'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:88: undefined reference to `rlTranslatef'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:89: undefined reference to `rlRotatef'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:91: undefined reference to `rlPushMatrix'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:92: undefined reference to `rlRotatef'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:93: undefined reference to `rlScalef'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:96: undefined reference to `rlPopMatrix'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:98: undefined reference to `rlRotatef'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:99: undefined reference to `rlTranslatef'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:100: undefined reference to `rlRotatef'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:101: undefined reference to `rlRotatef'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:102: undefined reference to `rlScalef'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:105: undefined reference to `rlPopMatrix'
CMakeFiles\models_rlgl_solar_system.dir/objects.a(models_rlgl_solar_system.c.obj): In function `DrawSphereBasic':
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:139: undefined reference to `rlBegin'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:140: undefined reference to `rlColor4ub'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:146: undefined reference to `rlVertex3f'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:149: undefined reference to `rlVertex3f'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:152: undefined reference to `rlVertex3f'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:156: undefined reference to `rlVertex3f'
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:159: undefined reference to `rlVertex3f'
CMakeFiles\models_rlgl_solar_system.dir/objects.a(models_rlgl_solar_system.c.obj):C:/projects/raylib/examples/models/models_rlgl_solar_system.c:162: more undefined references to `rlVertex3f' follow
CMakeFiles\models_rlgl_solar_system.dir/objects.a(models_rlgl_solar_system.c.obj): In function `DrawSphereBasic':
C:/projects/raylib/examples/models/models_rlgl_solar_system.c:167: undefined reference to `rlEnd'
collect2.exe: error: ld returned 1 exit status
examples\CMakeFiles\models_rlgl_solar_system.dir\build.make:87: recipe for target 'examples/models_rlgl_solar_system.exe' failed
mingw32-make.exe[2]: *** [examples/models_rlgl_solar_system.exe] Error 1
mingw32-make.exe[2]: Leaving directory 'C:/projects/raylib/build'
CMakeFiles\Makefile2:2246: recipe for target 'examples/CMakeFiles/models_rlgl_solar_system.dir/all' failed
mingw32-make.exe[1]: *** [examples/CMakeFiles/models_rlgl_solar_system.dir/all] Error 2
mingw32-make.exe[1]: Leaving directory 'C:/projects/raylib/build'
Makefile:161: recipe for target 'all' failed
mingw32-make.exe: *** [all] Error 2
Command exited with code 2
```

Can you take a look?

**Comment by myd7349** [comment]
@a3f This is strange. Cause `rlPushMatrix` is defined in a header file `rlgl.h`: https://github.com/raysan5/raylib/blob/master/src/rlgl.h#L988

**Comment by myd7349** [comment]
No I think the CI will pass. But I still do not know why.
Please do not merge yet, I want to dig it deeper.

**Comment by raysan5** [comment]
@myd7349 did you dig it? could we merge this PR?

**Comment by myd7349** [comment]
Sorry for the delay, I'm in bed now.  I will give it a test tomorrow.

**Comment by raysan5** [comment]
ok, no worries and no hurries! :)

**Comment by myd7349** [comment]
I added these lines to the end of `examples/CMakeLists.txt`:
```cmake
execute_process(
    COMMAND ${CMAKE_C_COMPILER} -E ${CMAKE_CURRENT_SOURCE_DIR}/models/models_rlgl_solar_system.c -I ${CMAKE_CURRENT_SOURCE_DIR}/others/external/include -I ${CMAKE_CURRENT_LIST_DIR}/../src
)
```
and noticed the preprocessed source file doesn't contain the definition of those functions mentioned in the error message.
Then I noticed a macro named `RLGL_IMPLEMENTATION` in `rlgl.h`.

To be continued...

**Comment by raysan5** [comment]
@myd7349 thank you very much for tracking this issue! I always use raylib as static library so I didn't though about it when adding `models_rlgl_solar_system.c` example...

When using raylib as shared library, probably rlgl functions should be included in the library, would it be enought just adding `RLAPI` definition same as `raylib.h`?

**Comment by raysan5** [comment]
`rlgl` modified on commit https://github.com/raysan5/raylib/commit/2249f123045208fa66c8c1c423020ffcf7eef284

**Comment by myd7349** [comment]
@raysan5 Thanks! I will give it a test later.

**Pull request merged**